### PR TITLE
Add middleware

### PR DESCRIPTION
**Pull request by chylli-deriv**

### CODE REVIEW

File: deriv_api/deriv_api.py [hunk 1]
```diff
@@ -246,14 +249,41 @@ async def send(self, request: dict) -> dict:
...
         response = await response_future
         self.cache.set(request, response)
         if self.storage:
             self.storage.set(request, response)
+        send_is_called = self._call_middleware('sendIsCalled',{'response': response, 'request': request})
+        if send_is_called:
+            return send_is_called
         return response
 
+    def _call_middleware(self, name: str, args: dict) -> Union[None,dict]:
```
Comment (shantanu-deriv) [inline deriv_api/deriv_api.py]: refactor it to the separate class?

Comment (chylli-deriv) [reply deriv_api/deriv_api.py]: How ? Could you please give me soem psudo-code ?

Comment (shantanu-deriv) [reply deriv_api/deriv_api.py]: I mean write a new middleware class and add this method there

Comment (chylli-deriv) [reply deriv_api/deriv_api.py]: ok got your idea.
Let me do it.
But when I implement this feature, in js version I didn't find any place that this feature is used. I suspect it is useless.  I even don't know I should keep this feature or not :(

Comment (shantanu-deriv) [reply deriv_api/deriv_api.py]: middlewares are useful when used stack-like manner, allowing you to perform actions downstream then filter and manipulate the response upstream. This means in a small workflow like this is not much need but keeping in mind the future, its okay

File: deriv_api/deriv_api.py [hunk 2]
```diff
@@ -38,7 +37,8 @@
     'deriv_api.deriv_api.DerivAPI.disconnect': False,
     'deriv_api.deriv_api.DerivAPI.send': False,
     'deriv_api.deriv_api.DerivAPI.wsconnection': False,
-    'deriv_api.deriv_api.DerivAPI.storage': False
+    'deriv_api.deriv_api.DerivAPI.storage': False,
+    'deriv_api.deriv_api.DerivAPI._call_middleware': False
```
Comment (raunakkathuria) [inline deriv_api/deriv_api.py]: Why the naming convention is different from others? Why not use `deriv_api.deriv_api.DerivAPI.middleware`

Comment (chylli-deriv) [reply deriv_api/deriv_api.py]: Sorry this is the old name that should be deleted. 
will drop it.

**Pull request merged**